test/parallel/test-net-connect-immediate-destroy.js

```diff
@@ -2,7 +2,11 @@
 const common = require('../common');
 const net = require('net');
 
-const socket = net.connect(common.PORT, common.localhostIPv4,
-                           common.mustNotCall());
-socket.on('error', common.mustNotCall());
-socket.destroy();
+const s = net.createServer();
+s.listen(0);
+const port = s.address().port;
+s.close(() => {
+  const socket = net.connect(port, common.localhostIPv4, common.mustNotCall());
+  socket.on('error', common.mustNotCall());
+  socket.destroy();
+});
```

Comment:
@santigimeno  Is that right though ? 
The original test tried to connect to `common.PORT` without creating a server thus ensuring that the connectListener `common.mustNotCall()` would not be invoked.
If i were to ensure that a server was listening on the port , that would invoke `common.mustNotCall()` => which is not what this test expects , right? 
(I'm trying to wrap my head around it, apologies if the questions seem silly)